### PR TITLE
Feature/mobile changes merge into develop

### PR DESCRIPTION
**Pull request by p-maks**
back end work related to mobile app

### CODE REVIEW

File: root/api/src/main/java/org/patientview/api/controller/UserController.java [hunk 1]
```diff
@@ -166,9 +166,21 @@ public void changePicture(@PathVariable("userId") Long userId, @RequestBody Stri
...
+                                                   @RequestBody SecretWordInput secretWordInput,
+                                                   @RequestParam(value = "salt", required = false) String salt)
             throws ResourceNotFoundException, ResourceForbiddenException {
-        userService.changeSecretWord(userId, secretWordInput);
+        return new ResponseEntity<>(
+                userService.changeSecretWord(userId, secretWordInput, StringUtils.isNotBlank(salt)),
+                HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/user/{userId}/checkSecretWord/{salt}", method = RequestMethod.POST)
```
Comment (jamesrossiter) [inline root/api/src/main/java/org/patientview/api/controller/UserController.java]: can you add javadoc for these methods if missing

File: root/api/src/main/java/org/patientview/api/service/AuthenticationService.java [hunk 2]
```diff
@@ -117,4 +128,14 @@ void checkSecretWord(org.patientview.persistence.model.User user, Map<String, St
...
+
+    /**
+     * Cleanup all the session tokens for the user except the current session.
+     *
+     * Used when user changes his password we need to invalidate all the session except the current one
+     *
+     * @param userId ID of the user to clean sessions for
+     */
+    @UserOnly
+    void cleanUpUserTokens(Long userId);
```
Comment (jamesrossiter) [inline root/api/src/main/java/org/patientview/api/service/AuthenticationService.java]: Should this be done as part of password reset in all instances (mobile, web etc)? If so can the functionality be included in the same call?

File: root/api/src/main/java/org/patientview/api/service/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -343,6 +346,151 @@ public Authentication authenticate(final Authentication authentication) throws A
...
+            }
+
+            // set temporary token
+            userToken.setSecretWordToken(CommonUtils.getAuthToken());
+            toReturn.setSecretWordToken(userToken.getSecretWordToken());
+
+            // set user token (must not be null)
+            userToken.setToken(CommonUtils.getAuthToken().substring(0, 40) + "secret");
+
+            // choose two characters to check and add to secret word indexes for ui
```
Comment (jamesrossiter) [inline root/api/src/main/java/org/patientview/api/service/impl/AuthenticationServiceImpl.java]: 3 characters for mobile

File: root/database/src/main/resources/db/migration/V1__Create_Schema.sql [hunk 4]
```diff
@@ -614,6 +614,8 @@ CREATE TABLE PV_Alert
   Web_Alert_Viewed            BOOLEAN      NOT NULL,
   Email_Alert                 BOOLEAN      NOT NULL,
   Email_Alert_Sent            BOOLEAN      NOT NULL,
+  Mobile_Alert                 BOOLEAN      NOT NULL,
```
Comment (jamesrossiter) [inline root/database/src/main/resources/db/migration/V1__Create_Schema.sql]: confirm these have been added to staging and production

Comment (p-maks) [inline root/database/src/main/resources/db/migration/V1__Create_Schema.sql]: Just Staging, Still needs to be done on Production

File: root/api/src/main/java/org/patientview/api/service/AuthenticationService.java (hunk 2 above)
Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/AuthenticationService.java]: No as we need token from security context, once they logged in, Updated docs

**Pull request merged**